tests/adapters/unified_tests/ip_management_test.py

```diff
@@ -22,92 +26,92 @@ class IpManagementTest(ConfiguredTestCase):
-        assert_that(response.status_code, equal_to(409))
+        with self.assertRaises(Conflict): #todo multiple exceptions: IPNotAvailable, IPAlreadySet
+            self.client.add_ip_to_vlan(vlan_number=2345, ip_network=IPNetwork("1.1.1.1/24"))
 
-        self.post("/switches/{switch}/vlans", data={"number": 2999})
-        response = self.post("/switches/{switch}/vlans/2999/ips", data={"address": "1.1.1.1", "mask": "24"}, fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(409))
+        self.client.add_vlan(number=2999)
+        with self.assertRaises(IPNotAvailable):
+            self.client.add_ip_to_vlan(vlan_number=2999, ip_network=IPNetwork("1.1.1.1/24"))
 
-        response = self.post("/switches/{switch}/vlans/2999/ips", data={"address": "1.1.1.2", "mask": "24"}, fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(409))
-        self.delete("/switches/{switch}/vlans/2999")
+        with self.assertRaises(IPNotAvailable):
+            self.client.add_ip_to_vlan(vlan_number=2999, ip_network=IPNetwork("1.1.1.2/24"))
 
-        response = self.delete("/switches/{switch}/vlans/1111/ips/1.1.1.1/24", fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(404)) #vlan not found
+        self.client.remove_vlan(number=2999)
 
-        response = self.delete("/switches/{switch}/vlans/2345/ips/2.2.2.2/24", fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(404)) #ip not found
+        with self.assertRaises(UnknownVlan):
+            self.client.remove_ip_from_vlan(vlan_number=1111, ip_network=IPNetwork("1.1.1.1/24"))
 
-        response = self.delete("/switches/{switch}/vlans/2345/ips/1.1.1.1/30", fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(404)) #ip not found
+        with self.assertRaises(UnknownIP):
+            self.client.remove_ip_from_vlan(vlan_number=2345, ip_network=IPNetwork("2.2.2.2/24"))
 
-        self.delete("/switches/{switch}/vlans/2345")
+        with self.assertRaises(UnknownIP):
+            self.client.remove_ip_from_vlan(vlan_number=2345, ip_network=IPNetwork("1.1.1.1/30"))
+
+        self.client.remove_vlan(2345)
 
     @skip_on_switches("juniper", "juniper_qfx_copper", "dell", "dell_telnet", "dell10g", "dell10g_telnet")
     def test_handling_access_groups(self):
-        self.post("/switches/{switch}/vlans", data={"number": 2345})
-
-        self.put("/switches/{switch}/vlans/2345/access-groups/in", raw_data="my-group")
-        self.put("/switches/{switch}/vlans/2345/access-groups/out", raw_data="your-group")
+        self.client.add_vlan(number=2345)
 
-        vlans = self.get("/switches/{switch}/vlans")
-        vlan2345 = next(vlan for vlan in vlans if vlan["number"] == 2345)
+        self.client.set_vlan_access_group(vlan_number=2345, direction=access_groups.IN, name="my-group")
```

Comment:
The access_groups pseudo enum is made to be used as symbols IN and OUT the module should not be imported